doc/man1/openssl-pkeyutl.pod.in

```diff
@@ -127,21 +132,25 @@ The input is a certificate containing a public key.
 =item B<-rev>
 
 Reverse the order of the input buffer. This is useful for some libraries
-(such as CryptoAPI) which represent the buffer in little endian format.
+(such as CryptoAPI) which represent the buffer in little-endian format.
 
 =item B<-sign>
 
-Sign the input data (which must be a hash) and output the signed result. This
-requires a private key.
+Sign the input data and output the signed result. This requires a private key.
+The input data given with the B<-in> option should be a hash value
+unless the use of a message digest operation is implied,
```

Comment:
This is not quite detailed enough IMHO. It seems to suggest that providing the input in the form of a  digest is the preferred/common use case. But that then drags in the need to e.g. use `-pkeyopt digest:<md>` in the RSA PKCS#1 case to get a valid signature.

It should say, that without `-rawin` (preferred) the input is assumed to already be a digest, but this may then require an additional `-pkeyopt digest:<md>` in some cases (e.g. RSA with the default PKCS#1 padding mode). Even with ECDSA (though the additional `-pkeyopt` option does not affect signature output, it is used to sanity-check that the input length is consistent with the purported digest).

Reply:
It is also unclear to me how this PR overlaps with the prior `-rawin` implication PR? Is one a merge-base for the other? Do they conflict?

I'd prefer to see a follow-on PR that has the prior as a merge-base, and worry about backports later once approved for master.

Reply:
Given your above text, I meanwhile understand better the use of `-pkeyopt digest:<md>`
and have added a fixup that integrates it (with minor adaptations).

Reply:
> It is also unclear to me how this PR overlaps with the prior `-rawin` implication PR? Is one a merge-base for the other? Do they conflict?

Yes - after merging #22910, there were conflicts.
 
> I'd prefer to see a follow-on PR that has the prior as a merge-base, 

This PR meanwhile is now based on the current master including the commits of #22910,
resolving the conflicts.

> and worry about backports later once approved for master.

Sounds good to me.